backend/app/controllers/spree/admin/promotions_controller.rb

```diff
@@ -2,35 +2,24 @@ module Spree
   module Admin
     class PromotionsController < ResourceController
       before_action :load_data
-      before_action :load_bulk_code_information, only: [:edit]
 
       helper 'spree/promotion_rules'
 
       def create
-        @promotion_builder = Spree::PromotionBuilder.new(
-          permitted_promo_builder_params.merge(user: try_spree_current_user),
-          permitted_resource_params
-        )
-        @promotion = @promotion_builder.promotion
+        @promotion = Spree::Promotion.new(permitted_resource_params)
+        @promotion.codes.new(value: params[:single_code]) if params[:single_code].present?
```

Comment:
I like how this separates creating a single code (via this controller) from creating a batch of codes (via the batch controller).  There's probably even further improvement we can do along the lines of creating single codes, but it seems like that could be a separate PR.  Possibly part of #1509?